packages/search-field/src/SearchField.test.js

```diff
@@ -295,20 +292,42 @@ describe("SearchField", () => {
     });
 
     test("should run onChange on Enter key with highlighted item", () => {
-      const index = 10;
+      const index = 0;
+      const value = 42;
+      const items = [{ value, label: "Foo" }];
+      const onChangeMock = jest.fn();
+
+      const component = mount(
+        <SearchField items={items} onChange={onChangeMock} />
+      );
       event.key = "Enter";
-      instance.state.highlightedIndex = index;
-      instance.onKeyDown(event);
+      component.setState({ highlightedIndex: index });
+      component.instance().onKeyDown(event);
 
-      expect(onChangeSpy).toHaveBeenCalledWith(index);
+      expect(onChangeMock).toHaveBeenCalledWith(value);
     });
 
-    test("should reset highlight on Esc key", () => {
+    test("should remove focus from the input on Enter key with highlighted item", () => {
+      const index = 0;
+      const value = 42;
+      const items = [{ value, label: "Foo" }];
+
+      const component = mount(<SearchField items={items} />);
+      event.key = "Enter";
+      component.setState({ highlightedIndex: index });
+      component.instance().onKeyDown(event);
+
+      expect(onBlurSpy).toHaveBeenCalled();
+    });
+    test("should reset highlight on Esc key", done => {
       event.key = "Escape";
       instance.onKeyDown(event);
       const stateChange = setStateSpy.mock.calls[0][0];
 
-      expect(stateChange).toHaveProperty("highlightedIndex", -1);
+      window.setTimeout(() => {
+        expect(stateChange).toHaveProperty("highlightedIndex", -1);
+        done();
+      }, 1000);
```

Comment:
Is this setTimeout necessary?
`stateChange` is set before the timeout, so why wait 1 second  to check it's properties?